include/utilities/span.hpp

```diff
@@ -0,0 +1,412 @@
+/*
+Copyright 2019-2023 Glen Joseph Fernandes
+(glenjofe@gmail.com)
+
+Distributed under the Boost Software License, Version 1.0.
+(http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+#if __has_include(<boost/core/span.hpp>)
+#include <boost/core/span.hpp>
+#else
+
+#ifndef BOOST_CORE_SPAN_HPP
+#define BOOST_CORE_SPAN_HPP
+
+#if __has_include(<boost/core/data.hpp>)
+#include <boost/core/data.hpp>
+#else
+#include "boostcore_data.hpp"
```

Comment:
Named like this to avoid tripping over the `.gitignore` line reading `boost_*`

Reply:
Hmm.... that should probably be `/boost_*` instead? Or `/boost_*/` if that works to match only directories? Purpose was to exclude boost distributions which may be untarred at the root of the repo. Probably should match only directories and probably only match at the root of the working copy. I'd consider this in-scope to fix if we think it's a better solution, which I'm thinking it might be.

Reply:
I don't expect the `data.hpp` header to ever be of interest to include on its own, so I don't think it's a big deal to just leave it. It's not going to make anything else worse, or otherwise cost us anything, I don't think.